### PR TITLE
[EASY] Bug: Count the protocol fees for the scoring in all the cases

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
We need to count the protocol fees for the scoring in all the cases. Good catch from @fleupold .

See @fleupold 's comment: https://github.com/cowprotocol/services/pull/2544#discussion_r1541118981

# Changes
- Count the protocol fees for the scoring in all the cases

## How to test
1. Regression tests

### CODE REVIEW

File: crates/driver/src/tests/cases/protocol_fees.rs [hunk 1]
```diff
@@ -174,7 +174,7 @@ async fn protocol_fee_calculated_on_the_solver_side() {
                 buy: 40.ether().into_wei(),
             },
         },
-        expected_score: 20.ether().into_wei(),
+        expected_score: 35.ether().into_wei(),
```
Comment (Mateo-mro) [inline crates/driver/src/tests/cases/protocol_fees.rs:177]: Now even if the `Solver` calculates the protocol fee, we still need to count it for the score, as per:

```rust
    /// CIP38 score defined as surplus + protocol fee
    ///
    /// Denominated in NATIVE token
    fn score(&self, prices: &auction::Prices) -> Result<eth::Ether, Error> {
        tracing::debug!("Scoring trade {:?}", self);
        Ok(self.native_surplus(prices)? + self.native_protocol_fee(prices)?)
    }
```

**Review by fleupold — APPROVED** [review]
Let's see what @squadgazzz and @sunce86 the masters of protocol fee related scoring say, but this looks correct to me.

File: crates/driver/src/tests/cases/protocol_fees.rs (hunk 1 above)
Comment (sunce86) [reply crates/driver/src/tests/cases/protocol_fees.rs:177]: Not too obvious from the numbers in the test that we get capped by volume, but the math seems correct.

**Review by sunce86 — APPROVED** [review]
Lg.

good catch @fleupold

File: crates/driver/src/tests/cases/protocol_fees.rs (hunk 1 above)
Comment (squadgazzz) [reply crates/driver/src/tests/cases/protocol_fees.rs:177]: Would it be possible to add comments to the test case explaining the expected values?

**Pull request merged**